### PR TITLE
feat: upgrade Bloom to Node 22, Next 15, and React 19

### PR DESCRIPTION
**Pull request by jaredcwhite**
This PR addresses #4730, #5397, #5449, #4328

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This is a full upgrade of our public/partners sites infrastructure. It doesn't address all tech debt (some notes below) but helps us stay current in terms of releases.

* Node 18 → **22**
* Next.js 14 → **15**
* React 18 → **19** 
* Sass 1.5 → **1.9**

and many additional dependencies thereof. You can see exact version numbers in the package/lock files.

The backend API is also running without modification now on Node 22. (There may be ideal upgrades to do there as well, but that's beyond the scope of this PR.)

Some upgrades I didn't pursue:

* Typescript got bumped up from 4.6.4 to **4.9.5**, but the current version is 5.9. I don't have a sense what's new/changed.
* React Hook Form is still at 6.15.8, but the current version is **7.68**.
* Tailwind CSS is still at 2.2.10. The latest 4.x line is a huge rewrite with many new features (could be interesting in fact: themes are now based on CSS variables, not JavaScript config, and it's a much faster Rust core!)
* We're still using old and busted Webpack, instead of new hotness Turbopack.
* We're not using any of the new "app router" functionality in Next.js, and continuing to use the "pages router". (Not necessarily a bad thing, from what I understand the app router rollout has been a slog, and then there was the latest critical vulnerability issue.)
* We're using some deprecated Sass conventions that we should try to move off of at some point. Search for `TODO: this is hiding tech debt!` to see the config I had to set for Sass so we wouldn't get a bunch of errors in the console.

## How Can This Be Tested/Reviewed?

All of the critical paths should be tested: signup/login/logout, viewing listings, submitting applications, partners creating/editing listings & applications, creating users, etc. We should be in decent shape based on my own QA and the test suites, but there's always a potential edge case to uncover.

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [x] Added tests covering the changes
- [x] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

**Comment by ludtkemorgan** [comment]
@jaredcwhite since this is a breaking change can you change the name of the PR to match that naming convention?

I'm still testing this, but if anyone else has issues you might need to delete your node modules directories and re install all packages.

File: sites/partners/package.json [hunk 1]
```diff
@@ -47,14 +47,16 @@
     "clone-deep": "^4.0.1",
     "dayjs": "^1.10.7",
     "dotenv": "^17.2.3",
-    "markdown-to-jsx": "^7.7.12",
+    "mapbox-gl": "^3.16.0",
+    "markdown-to-jsx": "^7.7.16",
     "nanoid": "^3.1.12",
-    "next": "^14.2.28",
+    "next": "^15.5.7",
```
Comment (ludtkemorgan) [inline sites/partners/package.json]: praise: This new feature showing at the bottom left of all pages in dev mode is nice 

Image

File: sites/public/src/components/listing/listing_sections/Availability.tsx [hunk 2]
```diff
@@ -171,7 +171,12 @@ export const Availability = ({ listing, jurisdiction }: AvailabilityProps) => {
         </Heading>
         {!alwaysShowStatusBar && subtitle && <p className={styles["bold-subheader"]}>{subtitle}</p>}
         {statusMessage && (
-          <p className={`${listingStyles["thin-heading-sm"]} seeds-m-bs-label`}>{statusMessage}</p>
+          <p
+            className={`${listingStyles["thin-heading-sm"]} seeds-m-bs-label`}
+            suppressHydrationWarning
```
Comment (emilyjablonski) [inline sites/public/src/components/listing/listing_sections/Availability.tsx]: **question:** J curious about this change!

File: sites/partners/src/components/listings/PaperListingForm/index.tsx [hunk 3]
```diff
@@ -112,6 +117,7 @@ const ListingForm = ({
   const defaultValues = editMode ? listing : formDefaults
   const formMethods = useForm<FormListing>({
     defaultValues,
+    mode: "onBlur",
```
[the remark this answers is not in the document]
Comment (jaredcwhite) [reply sites/partners/src/components/listings/PaperListingForm/index.tsx]: @emilyjablonski so I can't recall the exact rabbit trail that led me to this fix, but I was having some issues with validation working correctly on the partners listing form and this seemed to get us back to the previous behavior. I'm not sure why the default "onSubmit" wasn't working. Docs for context:

https://react-hook-form-website-git-leagcy-hook-form.vercel.app/v6/api/
